siliconcompiler/report/Home.py

```diff
@@ -0,0 +1,601 @@
+                                     active_color=active_toggle_color,
+                                     track_color=track_toggle_color)
+
+        streamlit.session_state['flowgraph'] = fg_toggle
+        if streamlit.session_state['flowgraph']:
+            streamlit.session_state['right after rerun'] = True
+            streamlit.experimental_rerun()
+
+    return None, metrics_and_nodes_info_col
+
+
+parser = argparse.ArgumentParser('dashboard')
+parser.add_argument('cfg', nargs='?')
+args = parser.parse_args()
+
+if not args.cfg:
+    raise ValueError('configuration not provided')
+
+title_col, job_select_col = streamlit.columns([0.7, 0.3], gap="large")
+
+if 'job' not in streamlit.session_state:
+    with open(args.cfg, 'r') as f:
+        config = json.load(f)
+
+    chip = Chip(design='')
+    chip.read_manifest(config["manifest"])
+    new_chip = chip
+    streamlit.session_state['master chip'] = chip
+    streamlit.session_state['job'] = 'default'
+else:
+    chip = streamlit.session_state['master chip']
+    new_chip = Chip(design='')
+    job = streamlit.session_state['job']
+    if job == 'default':
+        new_chip = chip
+    else:
+        new_chip.schema = chip.schema.history(job)
+        new_chip.set('design', chip.design)
+
+with title_col:
+    streamlit.title(f'{new_chip.design} dashboard')
+
+with job_select_col:
+    all_jobs = streamlit.session_state['master chip'].getkeys('history')
+    all_jobs.insert(0, 'default')
+    job = streamlit.selectbox(' ', all_jobs)
+    previous_job = streamlit.session_state['job']
+    streamlit.session_state['job'] = job
+    if previous_job != job:
+        streamlit.session_state['right after rerun'] = True
+        streamlit.experimental_rerun()
+
+# gathering data
+metric_dataframe = report.make_metric_dataframe(new_chip)
+
+# create mapping between task and step, index
+node_to_step_index_map = {}
+for step, index in metric_dataframe.columns.tolist():
+    node_to_step_index_map[step + index] = (step, index)
+
+# concatenate step and index
+metric_dataframe.columns = metric_dataframe.columns.map(lambda x:
+                                                        f'{x[0]}{x[1]}')
+nodes, edges = get_nodes_and_edges(new_chip,
+                                   report.get_flowgraph_edges(new_chip),
+                                   report.get_flowgraph_path(new_chip))
+manifest = report.make_manifest(new_chip)
+
+if 'flowgraph' not in streamlit.session_state:
+    streamlit.session_state['flowgraph'] = True
+
+if os.path.isfile(f'{new_chip._getworkdir()}/{new_chip.design}.png'):
+    tabs = streamlit.tabs(["Metrics",
+                           "Manifest",
+                           "File Preview",
+                           "Design Preview"])
+    metrics_tab, manifest_tab, file_preview_tab, design_preview_tab = tabs
+
+    with design_preview_tab:
+        streamlit.header('Design Preview')
+
+        streamlit.image(f'{new_chip._getworkdir()}/{new_chip.design}.png')
```

Comment:
what happens if this is missing? maybe check if this file is there?

Reply:
I do that a few lines earlier.